### PR TITLE
(uplift to 1.33.x) Retry signMessage for ConnectHardwareView (#11301)

### PR DESCRIPTION
**Pull request by spylogsster**
Uplift of #11301 #11332
Resolves https://github.com/brave/brave-browser/issues/19757
Resolves https://github.com/brave/brave-browser/issues/19824
Resolves https://github.com/brave/brave-browser/issues/19823

Should be merged to 1.33.x after https://github.com/brave/brave-core/pull/11317

Pre-approval checklist: 
- [x] You have tested your change on Nightly. 
- [ ] This contains text which needs to be translated. 
    - [ ] There are more than 7 days before the release. 
    - [ ] I've notified folks in #l10n on Slack that translations are needed. 
- [x] The PR milestones match the branch they are landing to. 

Pre-merge checklist: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 

Post-merge checklist: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Comment by kjozwiak** [comment]
Restarting all of the platforms as they've all failed due to the following:

```
[2021-11-30T14:06:38.770Z] HttpMethod: GET
[2021-11-30T14:06:38.771Z] URL: https://raw.githubusercontent.com/brave/brave-browser/pr11286_f/wallet/hw-flicker_1.33.x/package.json
[2021-11-30T14:06:38.771Z] Sending request to url: https://raw.githubusercontent.com/brave/brave-browser/pr11286_f/wallet/hw-flicker_1.33.x/package.json
[2021-11-30T14:06:38.972Z] Response Code: HTTP/1.1 404 Not Found
hudson.AbortException: Fail: Status code 404 is not in the accepted range: 100:399
```

**Review by onyb — APPROVED** [review]
Rebased on **`1.33.x`**.

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.33.x` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on Nightly as per https://github.com/brave/brave-core/pull/11301#issuecomment-983294626 & https://github.com/brave/brave-core/pull/11332#issuecomment-983288820.

**Pull request merged**
